This bug happen when I run attention_sink example's regression scripts. The  example_mha_sink_fwd_bhsd regression test report this bug:
```
/home/zyy/tilelang-opt/tilelang/examples/attention_sink# python regression_attention_sink.py 
Loading tilelang libs from dev root: /home/zyy/tilelang-opt/tilelang/build
  ├─ [1/6] example_gqa_sink_bwd_bhsd (24.10s)
  ├─ [2/6] example_gqa_sink_bwd_bhsd_sliding_window (17.93s)
  ├─ [3/6] example_mha_sink_bwd_bhsd (26.17s)
  ├─ [4/6] example_mha_sink_bwd_bhsd_sliding_window (21.08s)
  ├─ [5/6] example_mha_sink_fwd_bhsdTraceback (most recent call last):
  File "/home/zyy/tilelang-opt/tilelang/examples/attention_sink/regression_attention_sink.py", line 38, in <module>
    tilelang.testing.regression()
  File "/home/zyy/tilelang-opt/tilelang/tilelang/testing/perf_regression.py", line 103, in regression
    fn()
  File "/home/zyy/tilelang-opt/tilelang/examples/attention_sink/regression_attention_sink.py", line 8, in regression_example_mha_sink_fwd_bhsd
    tilelang.testing.process_func(example_mha_sink_fwd_bhsd.run_regression_perf)
  File "/home/zyy/tilelang-opt/tilelang/tilelang/testing/perf_regression.py", line 61, in process_func
    latency = float(func(**kwargs))
                    ^^^^^^^^^^^^^^
  File "/home/zyy/tilelang-opt/tilelang/examples/attention_sink/example_mha_sink_fwd_bhsd.py", line 267, in run_regression_perf
    kernel = flashattn(
             ^^^^^^^^^^
  File "/home/zyy/tilelang-opt/tilelang/tilelang/autotuner/tuner.py", line 1308, in __call__
    artifact = autotuner.run()
               ^^^^^^^^^^^^^^^
  File "/home/zyy/tilelang-opt/tilelang/tilelang/autotuner/tuner.py", line 994, in run
    jit_kernel = self.jit_compile()
                 ^^^^^^^^^^^^^^^^^^
  File "/home/zyy/tilelang-opt/tilelang/tilelang/autotuner/tuner.py", line 1292, in jit_compile
    return self.jit_impl(*args, **kwargs, __tune_params=config_arg)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/zyy/tilelang-opt/tilelang/tilelang/jit/__init__.py", line 461, in __call__
    kernel = self.compile(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/zyy/tilelang-opt/tilelang/tilelang/jit/__init__.py", line 391, in compile
    kernel_result = compile(
                    ^^^^^^^^
  File "/home/zyy/tilelang-opt/tilelang/tilelang/jit/__init__.py", line 111, in compile
    return cached(
           ^^^^^^^
  File "/home/zyy/tilelang-opt/tilelang/tilelang/cache/__init__.py", line 74, in cached
    return _dispatch_map[execution_backend].cached(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/zyy/tilelang-opt/tilelang/tilelang/cache/kernel_cache.py", line 363, in cached
    kernel = JITKernel(
             ^^^^^^^^^^
  File "/home/zyy/tilelang-opt/tilelang/tilelang/jit/kernel.py", line 131, in __init__
    adapter = self._compile_and_create_adapter(func, out_idx)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/zyy/tilelang-opt/tilelang/tilelang/jit/kernel.py", line 242, in _compile_and_create_adapter
    artifact = tilelang.lower(
               ^^^^^^^^^^^^^^^
  File "/home/zyy/tilelang-opt/tilelang/tilelang/engine/lower.py", line 330, in lower
    host_mod, device_mod, params, target, target_host = lower_to_host_device_ir(
                                                        ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/zyy/tilelang-opt/tilelang/tilelang/engine/lower.py", line 307, in lower_to_host_device_ir
    mod = OptimizeForTarget(mod, target)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/zyy/tilelang-opt/tilelang/tilelang/engine/phase.py", line 279, in OptimizeForTarget
    mod = tilelang.transform.SplitHostDevice()(mod)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/zyy/tilelang-opt/tilelang/3rdparty/tvm/python/tvm/ir/transform.py", line 171, in __call__
    return _ffi_transform_api.RunPass(self, mod)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "python/tvm_ffi/cython/function.pxi", line 968, in tvm_ffi.core.Function.__call__
  File "<unknown>", line 0, in tvm::transform::Pass::operator()(tvm::IRModule) const
  File "<unknown>", line 0, in tvm::transform::Pass::operator()(tvm::IRModule, tvm::transform::PassContext const&) const
  File "<unknown>", line 0, in tvm::transform::ModulePassNode::operator()(tvm::IRModule, tvm::transform::PassContext const&) const
  File "<unknown>", line 0, in std::_Function_handler<tvm::IRModule (tvm::IRModule, tvm::transform::PassContext), tvm::tl::transform::SplitHostDevice()::{lambda(tvm::IRModule, tvm::transform::PassContext)#1}>::_M_invoke(std::_Any_data const&, tvm::IRModule&&, tvm::transform::PassContext&&)
  File "<unknown>", line 0, in tvm::tl::transform::SplitHostDevice()::{lambda(tvm::IRModule, tvm::transform::PassContext)#1}::operator()(tvm::IRModule, tvm::transform::PassContext) const [clone .isra.0]
  File "<unknown>", line 0, in tvm::tl::SplitHostDevice(tvm::tirx::PrimFunc, tvm::IRModule*, std::function<tvm::GlobalVar ()>)
  File "<unknown>", line 0, in tvm::tirx::StmtMutator::VisitStmt(tvm::tirx::Stmt const&)
  File "<unknown>", line 0, in tvm::NodeFunctor<tvm::tirx::Stmt (tvm::ffi::ObjectRef const&, tvm::tirx::StmtFunctor<tvm::tirx::Stmt (tvm::tirx::Stmt const&)>*)>::operator()(tvm::ffi::ObjectRef const&, tvm::tirx::StmtFunctor<tvm::tirx::Stmt (tvm::tirx::Stmt const&)>*) const
  File "<unknown>", line 0, in tvm::tirx::StmtFunctor<tvm::tirx::Stmt (tvm::tirx::Stmt const&)>::InitVTable()::{lambda(tvm::ffi::ObjectRef const&, tvm::tirx::StmtFunctor<tvm::tirx::Stmt (tvm::tirx::Stmt const&)>*)#2}::_FUN(tvm::ffi::ObjectRef const&, tvm::tirx::StmtFunctor<tvm::tirx::Stmt (tvm::tirx::Stmt const&)>*)
  File "<unknown>", line 0, in tvm::tl::HostDeviceSplitter::VisitStmt_(tvm::tirx::AttrStmtNode const*)
  File "<unknown>", line 0, in tvm::tl::HostDeviceSplitter::SplitDeviceFunc(tvm::tirx::Stmt, tvm::Target)
  File "<unknown>", line 0, in tvm::tl::HostDeviceSplitter::SplitDeviceFunc(tvm::tirx::Stmt, tvm::Target)::{lambda()#1}::operator()() const
  File "<unknown>", line 0, in tvm::tirx::VarUseDefAnalyzer::VisitStmt_(tvm::tirx::AttrStmtNode const*)
  File "<unknown>", line 0, in tvm::tirx::StmtVisitor::VisitStmt_(tvm::tirx::SeqStmtNode const*)
  File "<unknown>", line 0, in tvm::tirx::VarUseDefAnalyzer::VisitStmt_(tvm::tirx::AttrStmtNode const*)
  File "<unknown>", line 0, in tvm::tirx::VarUseDefAnalyzer::VisitStmt_(tvm::tirx::AttrStmtNode const*)
  File "<unknown>", line 0, in tvm::tirx::VarUseDefAnalyzer::VisitStmt_(tvm::tirx::AttrStmtNode const*)
  [Previous line repeated 2 more times]
  File "<unknown>", line 0, in tvm::tirx::StmtVisitor::VisitStmt_(tvm::tirx::SeqStmtNode const*)
  File "<unknown>", line 0, in tvm::tirx::StmtVisitor::VisitStmt_(tvm::tirx::SeqStmtNode const*)
  File "<unknown>", line 0, in tvm::tirx::VarUseDefAnalyzer::VisitBufferDef(tvm::tirx::Buffer const&, bool)
  File "/home/zyy/tilelang-opt/tilelang/3rdparty/tvm/src/tirx/analysis/var_use_def_analysis.cc", line 167, in void tvm::tirx::VarUseDefAnalyzer::HandleDef(const tvm::tirx::Buffer&)
    TVM_FFI_ICHECK(!buffer_def_count_.count(ptr))
  
tvm.error.InternalError: Check failed: (!buffer_def_count_.count(ptr)) is false: buffer acc_s has already been defined, the Stmt is not SSA
```

--- a/src/transform/unroll_loop.cc
+++ b/src/transform/unroll_loop.cc
@@ -101,6 +101,51 @@ private:
 // The Visitor is used to check whether var is used as write index in a local
 // memory If a loop var is used as indices to a local memory, it must be
 // unrolled so the local memory access can be turned into register access.
+class UnrolledBodyDefFreshener : public StmtExprMutator {
+public:
+  PrimExpr VisitExpr_(const VarNode *op) final {
+    Var var = GetRef<Var>(op);
+    auto it = var_remap_.find(var);
+    if (it != var_remap_.end()) {
+      return (*it).second;
+    }
+    return var;
+  }
+
+  Buffer VisitBufferDef(const Buffer &buffer, bool alloc_data) final {
+    Var data;
+    if (alloc_data) {
+      data = FreshVar(buffer->data);
+    } else {
+      PrimExpr remapped_data = VisitExpr(buffer->data);
+      ICHECK(remapped_data.as<VarNode>())
+          << "Buffer data must remain a Var after freshening definitions";
+      data = Downcast<Var>(remapped_data);
+    }
+
+    auto visit_expr = [this](const PrimExpr &expr) {
+      return this->VisitExpr(expr);
+    };
+    Buffer new_buffer = buffer;
+    auto writer = new_buffer.CopyOnWrite();
+    writer->data = std::move(data);
+    writer->shape = buffer->shape.Map(visit_expr);
+    writer->strides = buffer->strides.Map(visit_expr);
+    writer->elem_offset = visit_expr(buffer->elem_offset);
+    buffer_remap_.Set(buffer, new_buffer);
+    return new_buffer;
+  }
+
+private:
+  Var FreshVar(const Var &var) {
+    Var new_var = Var(make_object<VarNode>(*var.get()));
+    var_remap_.Set(var, new_var);
+    return new_var;
+  }
+
+  Map<Var, Var> var_remap_;
+};
+
 class LoopUnroller : public StmtExprMutator {
 public:
   explicit LoopUnroller(int auto_max_step, int auto_max_depth,
@@ -244,6 +289,7 @@ public:
     for (int i = 0; i < value; ++i) {
       vmap.Set(op->loop_var, op->min + make_const(op->loop_var.dtype(), i));
       Stmt step = Substitute(body, vmap);
+      step = UnrolledBodyDefFreshener()(std::move(step));
       unrolled.push_back(step);
     }
     return SeqStmt::Flatten(unrolled);

--- a/testing/python/transform/test_tilelang_transform_unroll_loop.py
+++ b/testing/python/transform/test_tilelang_transform_unroll_loop.py
@@ -1,0 +1,31 @@
+import tilelang
+from tilelang import tvm
+from tvm.script import ir as I
+from tvm.script import tirx as T
+
+
+def test_unroll_decl_buffer_defs_are_fresh():
+    @I.ir_module
+    class Before:
+        @T.prim_func
+        def main(A: T.Buffer((16,), "float32")):
+            for i in T.unroll(2):
+                A_flat = T.decl_buffer((16,), "float32", data=A.data)
+                A_flat[0] = T.float32(i)
+
+    after = tilelang.transform.UnrollLoop()(Before)
+    body = after["main"].body
+
+    decls = [stmt for stmt in body.seq if isinstance(stmt, tvm.tirx.DeclBuffer)]
+    stores = [stmt for stmt in body.seq if isinstance(stmt, tvm.tirx.BufferStore)]
+
+    assert len(decls) == 2
+    assert len(stores) == 2
+    assert not decls[0].buffer.same_as(decls[1].buffer)
+    assert decls[0].buffer.data.same_as(decls[1].buffer.data)
+    assert stores[0].buffer.same_as(decls[0].buffer)
+    assert stores[1].buffer.same_as(decls[1].buffer)
+
+
+if __name__ == "__main__":
+    test_unroll_decl_buffer_defs_are_fresh()

